src/test/fuzz/merkleblock.cpp

```diff
@@ -16,12 +16,49 @@
 void test_one_input(const std::vector<uint8_t>& buffer)
 {
     FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
-    std::optional<CPartialMerkleTree> partial_merkle_tree = ConsumeDeserializable<CPartialMerkleTree>(fuzzed_data_provider);
-    if (!partial_merkle_tree) {
-        return;
+
+    {
+        std::optional<CPartialMerkleTree> partial_merkle_tree = ConsumeDeserializable<CPartialMerkleTree>(fuzzed_data_provider);
+        if (partial_merkle_tree) {
+            (void)partial_merkle_tree->GetNumTransactions();
+            std::vector<uint256> matches;
+            std::vector<unsigned int> indices;
+            while (fuzzed_data_provider.ConsumeBool()) {
+                indices.push_back(fuzzed_data_provider.ConsumeIntegral<unsigned int>());
+            }
+            (void)partial_merkle_tree->ExtractMatches(matches, indices);
+        }
+    }
+
+    {
+        CMerkleBlock merkle_block;
+        const std::optional<CBlock> opt_block = ConsumeDeserializable<CBlock>(fuzzed_data_provider);
+        CBloomFilter bloom_filter;
+        std::set<uint256> txids;
+        if (opt_block && !opt_block->vtx.empty()) {
+            if (fuzzed_data_provider.ConsumeBool()) {
+                merkle_block = CMerkleBlock{*opt_block, bloom_filter};
```

Comment:
Could a case be made to not only have the default, match-all filter?

Reply:
Fixed! I'm now covering all three:
* `CMerkleBlock{*opt_block, bloom_filter}`
* `CMerkleBlock{*opt_block, txids}`
* `CMerkleBlock{}` (added)